homework/src/app.js

```diff
@@ -2,10 +2,192 @@
   /* Your homework goes here */
   
   /* STRINGS */
+  /*1*/ 
+  const myString = "hello,this,is,a,difficult,to,read,sentence";
+  
+  console.log(myString.length);
+
+  console.log(myString.replace(/,/g ,' '));
+
+
+/* 2*/
+
+
+  const str = 'dlroW olleH';
+  const reverse = str 
+  console.log(Array.from(str).reverse().join(''));
 
   /* ARRAYS */
 
+  const favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+
+  favoriteAnimals.push('turtle');
+
+  // ["blowfish", "capricorn", "giraffe", "turtle"]
+
+  console.log(favoriteAnimals);
+
+  favoriteAnimals.splice(1,0,'horse');
+
+  // ["blowfish", "horse", "capricorn", "giraffe", "turtle"]
+
+  console.log(favoriteAnimals);
+
+  favoriteAnimals.splice(3,1,);
+
+  // ["blowfish", "horse", "capricorn", "turtle"]
+
+  console.log(favoriteAnimals);
+
+
   /* OPERATORS */
 
+  /* 1 */
+  
+  0 == '0'  // javascript converts number to string so string + string = true
+  0 == []   // 0 is falsy, [] is also falsy so falsy + falsy = true
+  "0" == [] // string is truthy. [] is falsy so truthy + falsy = false
+  
+  /* 2 */
+
+  const x = [1, 2, 3];
+  const y = [1, 2, 3];
+  const z = y;
+
+  console.log(x == y);  // false - reference is not same
+  console.log(x === y); // false - reference is not same
+  console.log(z === y); // true - objects returns true when they reference the same memory location.
+  console.log(z == x);  // false - reference is not same
+
+  /* 3 */
+
+  const o1 = { foo: "bar" };
+  const o2 = { foo: "bar" };
+  const o3 = o2;
+
+  console.log(o1 == o2);  // false
+  console.log(o2 == o3);  // true
+  console.log(o3 == 01);  // false
+  console.log(o2 == 01);  // false
+
+  // when we change o1 whit o3
+  /*
+  
+  const o3 = { foo: "bar" };
+  const o2 = { foo: "bar" };
+  const o1 = o2;
+  
+  */
+
+  console.log(o1 == o2);  // true
+  console.log(o2 == o3);  // false
+  console.log(o3 == 01);  // false
+  console.log(o2 == 01);  // false
+
+  // when we change o2 whit o3, giving syntax error
```

Comment:
changing `o2` should also change `o3` since they point to the same object.
for example:
`o2.foo = "abc";
console.log(o3);`
should print `abc`.

Reply:
you mean, if we reassign o2, this changing will affect result of o3,right?